gradio/interface.py

```diff
@@ -213,17 +213,30 @@ def __init__(
         else:
             self.cache_examples = cache_examples or False
 
-        if "state" in inputs or "state" in outputs:
-            state_input_count = len([i for i in inputs if i == "state"])
-            state_output_count = len([o for o in outputs if o == "state"])
-            if state_input_count != 1 or state_output_count != 1:
-                raise ValueError(
-                    "If using 'state', there must be exactly one state input and one state output."
-                )
-            default = utils.get_default_args(fn)[inputs.index("state")]
-            state_variable = Variable(value=default)
-            inputs[inputs.index("state")] = state_variable
-            outputs[outputs.index("state")] = state_variable
+        state_input_indexes = [
+            idx for idx, i in enumerate(inputs) if i == "state" or isinstance(i, State)
+        ]
+        state_output_indexes = [
+            idx for idx, o in enumerate(outputs) if o == "state" or isinstance(o, State)
+        ]
+
+        if len(state_input_indexes) == 0 and len(state_output_indexes) == 0:
+            pass
+        elif len(state_input_indexes) != 1 or len(state_output_indexes) != 1:
+            raise ValueError(
+                "If using 'state', there must be exactly one state input and one state output."
+            )
+        else:
+            state_input_index = state_input_indexes[0]
+            state_output_index = state_output_indexes[0]
+            if inputs[state_input_index] == "state":
+                default = utils.get_default_args(fn)[state_input_index]
```

Comment:
Unrelated, but the type hint for `get_default_args` is wrong (Dict instead of List)

Reply:
Fixed